### ISSUE TITLE
Idea: quick fix to disable rule for selected line(s)

### ISSUE THREAD

**Issue by OliverJAsh** [microsoft/vscode-eslint#245]
I make frequent use of comments to disable selected ESLint rules for selected lines, for example:

``` js
// eslint-disable-next-line max-len
{} // long line of JavaScript …
```

It would be great if VSCode provided a "quick fix" option in the UI to silence rules it is complaining about by automatically adding these comments for the selected lines. I get quite tired of writing them, especially looking up the precise rule names.

Has this idea come up before?

**Comment by dbaeumer** [comment]
Would be a good opportunity for a PR

**Comment by wmertens** [comment]
Yes, I often wish for this too. There could be a button on the tooltip "disable for this line".

I'd love to help but don't know where to start. Basically there should be two commands, `disable-eslint-linebefore` and `disable-eslint-sameline`, and they should insert or amend the currently hovered or cursored error into the editor.

**Comment by fiznool** [comment]
Would love to see this too. The same feature is currently available in WebStorm:

<img width="501" alt="screen shot 2018-01-11 at 10 07 19" src="https://user-images.githubusercontent.com/820863/34819994-4155360a-f6b7-11e7-9256-3cddabfcbbe2.png">

**Comment by agrande** [comment]
Is anybody working on this?
I switched over from WebStorm to VSCode recently, and I miss that feature very much.

**Comment by dbaeumer** [comment]
Not that I am aware of. I can point you to the right code locations in case you want to give it a try.

**Comment by agrande** [comment]
Yes please, I might do just that, if it’s not too complex (I’m new to VSCode, I’d have to read up a bit about developing extensions).
The way I see it, that’d be a couple of additional entries in the Quickfix command contextual menu to suppress the error/warning for the line or the whole file (or even the eslint config file).

**Comment by dbaeumer** [comment]
@agrande that is correct, however the fixes need to be contribute in the ESLint linting server code. We run the linters outside of VS Code itself in its own process. The corresponding code is here: https://github.com/Microsoft/vscode-eslint/blob/master/server/src/eslintServer.ts#L1015

So instead of only adding the fixes we would need to:

- have commands for fixes that add an ignore comment.
- add code action results for these commands.

**Comment by bendtherules** [comment]
Just a temp solution I use -
Add eslint disable snippets as user snippet in VSCode

```json
{
	"Disable eslint rule next line": {
		"prefix": "eslint-disable-next",
		"body": [
			"eslint-disable-next-line ${1:<rule-name>}"
		],
		"description": "Disable eslint rules for the next line \n (using inline syntax)"
	},
	"Disable eslint rule current line": {
		"prefix": "eslint-disable-current",
		"body": [
			"eslint-disable-line ${1:<rule-name>}"
		],
		"description": "Disable eslint rules for the current line \n (using inline syntax)"
	}
}
```

To add - 
1. Preferences - Configure User Snippets
2. Choose Javascript
3. Paste above snippet

**Comment by overlookmotel** [comment]
I think this can be closed since feature was added in #530.
